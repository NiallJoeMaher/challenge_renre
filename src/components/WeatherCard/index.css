.c-WeatherCard {
  padding: 20px;
  border-radius: 4px;
  background: #393f4d;
  box-shadow: 0px 0px 30px 3px rgba(0,0,0,0.75);
}

.c-WeatherCard__value {
  color: #feda6a;
  font-family: 'Fredericka the Great', cursive;
  font-size: 30px;
  padding: 8px 16px;
}

.c-WeatherCard__value-text {
  font-size: 16px;
  font-family: 'Josefin Slab', serif;
  color: #d4d4dc;
}
