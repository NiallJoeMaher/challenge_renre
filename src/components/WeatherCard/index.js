import React, { Component } from 'react';
import './index.css';

import PropTypes from 'prop-types';

class WeatherCard extends Component {
  static propTypes = {
    date: PropTypes.string.isRequired,
    highValue: PropTypes.string.isRequired,
    lowValue: PropTypes.string.isRequired,
    day: PropTypes.string.isRequired
  };

  render() {
    const className = 'c-WeatherCard';
    const { date, highValue, lowValue, day } = this.props;
    return (
      <div className={className}>
        <div className={`${className}__date`}>{`${day} ${date}`}</div>
        <div className={`${className}__value`}>
          <span className={`${className}__value-text`}>Highs of </span>
          &deg;{highValue}
        </div>
        <div className={`${className}__value`}>
          <span className={`${className}__value-text`}>Lows of </span>
          &deg;{lowValue}
        </div>
      </div>
    );
  }
}

export default WeatherCard;
