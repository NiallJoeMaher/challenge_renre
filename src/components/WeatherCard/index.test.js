import React from 'react';
import { shallow } from 'enzyme';

import WeatherCard from './';

it('renders as expected', () => {
  const component = shallow(
    <WeatherCard date='26 Jan 2018' day='Thur' highValue='100' lowValue='1' />
  );
  expect(component).toMatchSnapshot();
});
