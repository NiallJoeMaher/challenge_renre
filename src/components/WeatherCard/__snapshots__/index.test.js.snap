// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="c-WeatherCard"
>
  <div
    className="c-WeatherCard__date"
  >
    Thur 26 Jan 2018
  </div>
  <div
    className="c-WeatherCard__value"
  >
    <span
      className="c-WeatherCard__value-text"
    >
      Highs of 
    </span>
    °
    100
  </div>
  <div
    className="c-WeatherCard__value"
  >
    <span
      className="c-WeatherCard__value-text"
    >
      Lows of 
    </span>
    °
    1
  </div>
</div>
`;
